src/core/oauth2-authorize.js

```diff
@@ -34,14 +34,44 @@ export default function authorize ( auth, authActions, errActions, configs ) {
       + "&state=" + encodeURIComponent(state)
       + "&client_id=" + encodeURIComponent(clientId)
 
-  // pass action authorizeOauth2 and authentication data through window
-  // to authorize with oauth2
-  win.swaggerUIRedirectOauth2 = {
-    auth: auth,
-    state: state,
-    callback: authActions.preAuthorizeOauth2,
-    errCb: errActions.newAuthErr
-  }
+  if (flow === "application") {
+    fetch(schema.get("tokenUrl"), {
```

Comment:
Please move this to actions like for password flow 
https://github.com/swagger-api/swagger-ui/blob/master/src/core/oauth2-authorize.js#L12-L15

Actions: https://github.com/swagger-api/swagger-ui/blob/master/src/core/plugins/auth/actions.js